subprojects/ide-provisioning/src/main/java/org/gradle/profiler/ide/idea/IDEAProvider.java

```diff
@@ -0,0 +1,74 @@
+package org.gradle.profiler.ide.idea;
+
+import com.intellij.ide.starter.community.PublicIdeDownloader;
+import com.intellij.ide.starter.community.model.BuildType;
+import com.intellij.ide.starter.ide.IdeArchiveExtractor;
+import com.intellij.ide.starter.ide.IdeDownloader;
+import com.intellij.ide.starter.ide.installer.IdeInstallerFile;
+import com.intellij.ide.starter.models.IdeInfo;
+import org.gradle.profiler.ide.IdeProvider;
+import org.jetbrains.annotations.Nullable;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Collections;
+
+public class IDEAProvider implements IdeProvider<IDEA> {
+    private final IdeDownloader ideDownloader;
+    private final IdeArchiveExtractor ideArchiveExtractor;
+
+    public IDEAProvider() {
+        this.ideDownloader = PublicIdeDownloader.INSTANCE;
+        this.ideArchiveExtractor = IdeArchiveExtractor.INSTANCE;
+    }
+
+    @Override
+    public File provideIde(IDEA ide, Path homeDir, Path downloadsDir) {
+        IdeInfo ideInfo = new IdeInfo(
+            "IC",
+            "Idea",
+            "idea",
+            BuildType.EAP.getType(),
+            Collections.emptyList(),
+            "", // latest
+            ide.getVersion(),
+            null,
+            null,
+            "IDEA Community",
+            info -> null
+        );
+
+        String version = ideInfo.getVersion().isEmpty()
```

Comment:
Here and below own cache logic was implemented. Why:
IdeDownloader is able to skip downloading, but it's considering a presence of installer(like a `.dmg`) file.
However, I'm deleting an installer after unpacking of it, because it's quite big (1+ Gb).
At the same time, I don't want to download IDE each time. So simple heuristics was introduced - create new dir like `ideaIC/latest/` and assuming that only one file (ide installation, in fact, like a `.app`) is there. If it there - skip downloading.

Reply:
Doesn't idea starter do any caching?

Reply:
It does. But it's relying on presence of installer(like a `.dmg` file). If we want to delete installers after unpacking(I guess we want), then we can't rely on it